### PR TITLE
fix: channels creation on migrate

### PR DESCRIPTION
**Pull request by ruben-arts**
Now the migration script will result in:

```toml
[envs.package2]
channels = ["file:///C:/Users/Ruben/development/pixi/tests/integration/test_data/global_update_channel_1/output"]
dependencies = { package2 = "*" }
exposed = { package2 = "package2" }

[envs.pixi-pack]
channels = ["conda-forge"]
dependencies = { pixi-pack= "*" }
exposed = { pixi-pack= "pixi-pack" }
```` 
Instead of 
```toml
[envs.package2]
channels = ["file:///C:/Users/Ruben/development/pixi/tests/integration/test_data/global_update_channel_1/output"]
dependencies = { package2 = "*" }
exposed = { package2 = "package2" }

[envs.pixi-pack]
channels = ["https://conda.anaconda.orge/conda-forge/"]
dependencies = { pixi-pack= "*" }
exposed = { pixi-pack= "pixi-pack" }
````

This PR also fixes the migration from multi channel environments, as it was not checking all channels used in the initial install.

Before:
```toml
[envs.lsd]
channels = ["https://prefix.dev/rust-forge"]
dependencies = { lsd = "*" }
exposed = { lsd = "lsd" }
```

after this PR:
```toml
[envs.lsd]
channels = [
    "https://prefix.dev/rust-forge",
    "conda-forge",
]
dependencies = { lsd = "*" }
exposed = { lsd = "lsd" }
```

### CODE REVIEW

**Comment by ruben-arts** [comment]
Made it draft as I'm going to make this a function for later reuse, spoiler we currently don't support multi channel environments.

**Comment by Hofer-Julian** [comment]
> spoiler we currently don't support multi channel environments.

Oh no. In general or just for migration?

**Comment by ruben-arts** [comment]
@Hofer-Julian that is now also fixed. It was only an issue on migration.

File: tests/integration/test_global.py [hunk 1]
```diff
@@ -161,10 +161,18 @@ def test_sync_migrate(pixi: Path, tmp_path: Path, dummy_channel_1: str) -> None:
...
     toml = f"""\
+# Test with special channel
 [envs.test]
 channels = ["{dummy_channel_1}"]
 dependencies = {{ dummy-a = "*", dummy-b = "*" }}
 exposed = {{ dummy-1 = "dummy-a", dummy-2 = "dummy-a", dummy-3 = "dummy-b", dummy-4 = "dummy-b" }}
+
+# Test with conda-forge channel
+[envs.test2]
+channels = ["conda-forge"]
```
Comment (Hofer-Julian) [inline tests/integration/test_global.py:172]: Would it make sense to add a `dummy-channel` here to make sure that everything also works with multiple channels within one environment?

Comment (ruben-arts) [reply tests/integration/test_global.py:180]: Thanks for doing it! @Hofer-Julian

**Review by Hofer-Julian — APPROVED** [review]

**Pull request merged**